### ISSUE TITLE
Does not use DataType.Password

### ISSUE THREAD

**Issue by vanillajonathan** [domaindrivendev/Swashbuckle.AspNetCore#521]
Swashbuckle does not set the input type to password when the property is decorated using the `[DataType(DataType.Password)]` property decorator.

```cs
    public class Model
    {
        [Required]
        [DataType(DataType.Password)]
        public string Password { get; set; }
    }
```

**Comment by funkycoding** [comment]
As you figured out, Swashbuckle doesn't use the DataType attributes. @domaindrivendev is there a need / wish to support this?

You could use a custom IOperationFilter to get what you want, something like this:

```
public class PasswordOperationFilter : IOperationFilter
{
    private static Type _modelType = typeof(Address);

    public void Apply(Operation operation, OperationFilterContext context)
    {
        if(context.ApiDescription.ActionDescriptor.Parameters.Any(x => x.ParameterType == _modelType))
        {
            var password = operation.Parameters.OfType<NonBodyParameter>()
                                               .SingleOrDefault(x => x.Name == nameof(Model.Password));
            if (password != null)
            {
                password.Format = "password";
            }
        }
    }
}
```

Of course, you can make this more generic and have it work on the DataType attribute.

**Comment by vanillajonathan** [comment]
It would be nice if Swashbuckle did this out-of-the-box.

**Comment by domaindrivendev** [comment]
Addressed by 7e762ef41d7c6b0a82c0cddfca6e360355b379fa.
